internal/audit/cerboshub/cerboshub.go

```
@@ -0,0 +1,237 @@
+// Copyright 2021-2024 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package cerboshub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	auditv1 "github.com/cerbos/cerbos/api/genpb/cerbos/audit/v1"
+	"github.com/cerbos/cerbos/internal/audit"
+	"github.com/cerbos/cerbos/internal/audit/local"
+	"github.com/cerbos/cerbos/internal/config"
+	badgerv4 "github.com/dgraph-io/badger/v4"
+	"github.com/dgraph-io/ristretto/z"
+	"go.uber.org/zap"
+)
+
+const (
+	goroutineResetThreshold = 1 << 16
+
+	Backend = "cerboshub"
+)
+
+var (
+	SyncStatusPrefix   = []byte("bs") // "b" for contiguity with audit log keys in LSM, "s" because "sync"
+	AccessSyncPrefix   = []byte("bsacc")
+	DecisionSyncPrefix = []byte("bsdec")
+)
+
+func init() {
+	audit.RegisterBackend(Backend, func(_ context.Context, confW *config.Wrapper, decisionFilter audit.DecisionLogEntryFilter) (audit.Log, error) {
+		conf := new(Conf)
+		if err := confW.GetSection(conf); err != nil {
+			return nil, fmt.Errorf("failed to read cerboshub audit log configuration: %w", err)
+		}
+
+		syncer := NewIngestSyncer()
+
+		return NewLog(conf, decisionFilter, syncer)
```

Let's return an error here until we are ready to release this feature.